pkgs/top-level/perl-packages.nix

```diff
@@ -8948,12 +8948,21 @@ let self = _self // overrides; _self = with self; {
       sha256 = "dda2578d7b32152c4afce834761a61d117de286c705a9f7972c7ac6032ca5953";
     };
     propagatedBuildInputs = [ FileListing HTMLParser HTTPCookies HTTPDaemon HTTPNegotiate NetHTTP TryTiny WWWRobotRules ];
+    buildInputs = [ TestFatal TestNeeds TestRequiresInternet ];
+
+    # It fixes cross-compilation error by commenting out `has_module` call:
+    # > Can't load module B, dynamic loading not available in this perl.
+    # > (You may need to build a new perl executable which either supports
+    # > dynamic loading or has the B module statically linked into it.)
+    preConfigure = ''
+      sed -r -i "s@(requires\('HTTP::Status', '[0-9.]+'\)) if has_module@\\1; # @" Makefile.PL
```

Comment:
This code is auto-generated. Is this a fundamental flaw of the perl build systems?

Reply:
The problem is attempt to dynamically load ```.so``` for wrong architecture.
That ```.so``` could be linked statically, it would solve the problem with ```perlPackages.LWP``` but there are few other modules (for example ```perlPackages.JSON```) which will still require a patch

Reply:
cross-compilation ```perlPackages.XMLParser``` is also broken, for the same but in a bit different way, which makes creation of a generic solution difficult

```
ERROR from evaluation of /build/XML-Parser-2.44/Expat/Makefile.PL: Can't load module Tie::Hash::NamedCapture, dynamic loading not available in this perl. 
  (You may need to build a new perl executable which either supports                                                                                      
  dynamic loading or has the Tie::Hash::NamedCapture module statically linked into it.)                                                                   
 at /nix/store/b6w2xp3hbq7bgdgxy2dp48ydf68l6vvj-perl-5.28.0-aarch64-unknown-linux-gnu/lib/perl5/5.28.0/English.pm line 155.                               
```

Reply:
```LD_LIBRARY_PATH``` with all host-architecture ```.so``` might be a generic solution...